library/spdm_crypt_lib/libspdm_crypt_asym.c

```diff
@@ -589,12 +589,49 @@ bool libspdm_sm2_dsa_verify_wrap (void *context, size_t hash_nid,
 }
 #endif
 
+static void libspdm_copy_signature_swap_endian(
+    uint8_t *endian_swapped_signature_buffer,
+    size_t endian_swapped_signature_buffer_size,
+    bool is_dual_buffer,
```

Comment:
`is_dual_buffer` brings confusion to me, and it takes me sometime to understand what it really means.


Maybe just define two functions: libspdm_copy_signature_swap_endian_for_ecdsa() and libspdm_copy_signature_swap_endian_for_rsa() ?

Then those functions can be reused by libspdm_asym_signature_swap_endian_if_necessary().

Reply:
Agree. I split into rsa and ecdsa versions and tried to make the code clearer.